### PR TITLE
FakeItEasy mock samples

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Not an exhaustive list but covers core groups of related types.

For *Properties.[Etag]/[LastModified] used reflection, which is a gap and documented in the helper.

### CODE REVIEW

File: Microsoft.Azure.Cosmos.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27130.2024
+# Visual Studio Version 16
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.sln]: Won't this conflict with your other editorconfig PR?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy/ContainerUT.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
+﻿namespace Microsoft.Azure.Cosmos.UT.FakeItEasy
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy/ContainerUT.cs]: Would it make sense to be consistent with the other test project regarding file naming? As in `ContainerTests`?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy.csproj [hunk 3]
```diff
@@ -0,0 +1,30 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+    
+    <!--<DirectVersion>3.1.3</DirectVersion>
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy.csproj]: Leftover from test?

File: Microsoft.Azure.Cosmos.sln (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.sln]: Yes it does. Its a old PR :-) will fix it.

**Comment by kirankumarkolli** [comment]
Lets chat more tomorrow.
1. UT vs Samples/Usage
2. Naming

**Comment by kirankumarkolli** [comment]
Lets move these two samples/usage.

**Comment by ryancrawcour** [comment]
i see this was last worked on Aug 15, it has since been approved, but has not yet been merged. Could we please get this merged as these samples would be VERY useful for everyone working with this SDK.

**Comment by SarahSexton** [comment]
> i see this was last worked on Aug 15, it has since been approved, but has not yet been merged. Could we please get this merged as these samples would be VERY useful for everyone working with this SDK.

Cosmos DB SDK Team: Please bring some attention back to this 🙏🏼

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy/ContainerUT.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            Container container = A.Fake<Container>();
+            ContainerResponse fakeResponse = A.Fake<ContainerResponse>();
+
+            A.CallTo(() => container.ReadContainerAsync(A<ContainerRequestOptions>.Ignored, A<CancellationToken>.Ignored))
+                .Returns(fakeResponse);
+
+            ContainerResponse response = await container.ReadContainerAsync();
+
+            Assert.ReferenceEquals(fakeResponse, response);
+            A.CallTo(() => container.ReadContainerAsync(A<ContainerRequestOptions>.Ignored, A<CancellationToken>.Ignored))
```
Comment (simplynaveen20) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.UT.FakeItEasy/Microsoft.Azure.Cosmos.UT.FakeItEasy/ContainerUT.cs:32]: Are we not asserting obvious here , we called ReadContainerAsync above. Instead faking some internal method on api and then asserting ?

**Comment by j82w** [comment]
Closing since there has been no update for over 30 days. Please re-open if necessary.

**Pull request closed without merging**